lib/interface_quda.cpp

```diff
@@ -1842,6 +1872,457 @@ void dslashQuda(void *h_out, void *h_in, QudaInvertParam *inv_param, QudaParity
+
+  pushVerbosity(inv_param.verbosity);
+  if (getVerbosity() >= QUDA_DEBUG_VERBOSE) printQudaInvertParam(&inv_param);
+
+  ColorSpinorParam cpuParam(h_in, inv_param, gauge.X(), false, inv_param.input_location);
+  ColorSpinorField *in_h = ColorSpinorField::Create(cpuParam);
+  ColorSpinorParam cudaParam(cpuParam, inv_param, QUDA_CUDA_FIELD_LOCATION);
+
+  cpuParam.v = h_out;
+  cpuParam.location = inv_param.output_location;
+  ColorSpinorField *out_h = ColorSpinorField::Create(cpuParam);
+
+  cudaParam.create = QUDA_NULL_FIELD_CREATE;
+  ColorSpinorField in(cudaParam); // cudaColorSpinorField
+  in = *in_h;
+  ColorSpinorField out(cudaParam); // cudaColorSpinorField
+  out = in;
+  ColorSpinorField tmp(cudaParam); // cudaColorSpinorField
+  tmp = in;
+
+  profileCovDev.TPSTOP(QUDA_PROFILE_INIT);
+
+  profileCovDev.TPSTART(QUDA_PROFILE_H2D);
+  in = *in_h;
+  profileCovDev.TPSTOP(QUDA_PROFILE_H2D);
+
+  profileCovDev.TPSTART(QUDA_PROFILE_COMPUTE);
+
+  if (getVerbosity() >= QUDA_DEBUG_VERBOSE) {
+    double cpu = blas::norm2(*in_h);
+    double gpu = blas::norm2(in);
+    printfQuda("In CPU %e CUDA %e\n", cpu, gpu);
+  }
+
+  inv_param.dslash_type = QUDA_COVDEV_DSLASH; // ensure we use the correct dslash
+  DiracParam diracParam;
+  setDiracParam(diracParam, &inv_param, false);
+
+  GaugeCovDev *myCovDev = new GaugeCovDev(diracParam); // create the Dirac operator
+
+  if (sym & 1) {
+    myCovDev->MCD(out, in, dir);                // apply the operator
+  }  if (sym & 2) {
+    myCovDev->MCD(tmp, in, dir+4);              // apply the operator
+  }
+
+  quda::blas::xpy(tmp, out);
+
+  if (sym == 3)
+    quda::blas::ax(0.5, out);
+
+  profileCovDev.TPSTOP(QUDA_PROFILE_COMPUTE);
+
+  profileCovDev.TPSTART(QUDA_PROFILE_D2H);
+  *out_h = out;
+  profileCovDev.TPSTOP(QUDA_PROFILE_D2H);
+
+  if (getVerbosity() >= QUDA_DEBUG_VERBOSE) {
+    double cpu = blas::norm2(*out_h);
+    double gpu = blas::norm2(out);
+    printfQuda("Out CPU %e CUDA %e\n", cpu, gpu);
+  }
+
+  profileCovDev.TPSTART(QUDA_PROFILE_FREE);
+  delete myCovDev; // clean up
+
+  delete out_h;
+  delete in_h;
+  profileCovDev.TPSTOP(QUDA_PROFILE_FREE);
+
+  popVerbosity();
+  profileCovDev.TPSTOP(QUDA_PROFILE_TOTAL);
+}
+
+void spinTasteQuda(void *h_out, void *h_in, int spin_, int taste, QudaInvertParam inv_param)
+{
+  profileCovDev.TPSTART(QUDA_PROFILE_TOTAL);
+  profileCovDev.TPSTART(QUDA_PROFILE_INIT);
+
+  const auto &gauge = *gaugePrecise; //(inv_param->dslash_type != QUDA_ASQTAD_DSLASH) ? *gaugePrecise : *gaugeFatPrecise;
+
+  inv_param.solution_type = QUDA_MAT_SOLUTION;
+  inv_param.dirac_order = QUDA_DIRAC_ORDER;
+
+  if (!gaugePrecise)
+    errorQuda("Gauge field not allocated");
+
+  pushVerbosity(inv_param.verbosity);
+  if (getVerbosity() >= QUDA_DEBUG_VERBOSE) printQudaInvertParam(&inv_param);
+
+  ColorSpinorParam cpuParam(h_in, inv_param, gauge.X(), false, inv_param.input_location);
+  ColorSpinorField *in_h = ColorSpinorField::Create(cpuParam);
```

Comment:
Move to stack

Reply:
fixed in 4fa8f65983905f955fa1dabad997b25e447653df